immuneML/preprocessing/filters/ChainRepertoireFilter.py

```diff
@@ -40,9 +38,9 @@ def process_dataset(self, dataset: RepertoireDataset, result_path: Path = None):
 
     @staticmethod
     def process(dataset: RepertoireDataset, params: dict) -> RepertoireDataset:
-        Preprocessor.check_dataset_type(dataset, [RepertoireDataset], "ChainRepertoireFilter")
+        ChainRepertoireFilter.check_dataset_type(dataset, [RepertoireDataset], "ChainRepertoireFilter")
         processed_dataset = dataset.clone()
-        PathBuilder.build(params["result_path"])
```

Comment:
yes, building the path has been moved to Filter.build_new_metadata  (which was a bugfix for another preprocessor, but better to have it there anyway because the metadata was the reason for needing to build the path)